shoes-core/lib/shoes/rect.rb

```diff
@@ -8,14 +9,9 @@ class Rect
     style_with :angle, :art_styles, :curve, :common_styles, :dimensions
     STYLES = { angle: 0 }
 
-    def initialize(app, parent, left, top, width, height, styles = {}, blk = nil)
-      @app = app
-      @parent = parent
-      style_init styles
+    def create_dimensions(args)
+      left, top, width, height = args
```

Comment:
As long as args has the right number of arguments for any given DSL class, I think you might be right. Will experiment with it and report back.